src/ui/components/pages/importEmail/signContext.tsx

```diff
@@ -0,0 +1,161 @@
+import clsx from 'clsx';
+import { LedgerConnectModal } from 'ledger/connectModal';
+import { ledgerService, LedgerServiceStatus } from 'ledger/service';
+import { usePopupSelector } from 'popup/store/react';
+import {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button } from 'ui/components/ui/buttons/Button';
+import { Modal } from 'ui/components/ui/modal/Modal';
+import Background from 'ui/services/Background';
+
+import { Login } from './login';
+import * as styles from './signWrapper.module.css';
+
+export const SignContext = createContext<{
+  openConfirmModal: () => Promise<unknown>;
+  closeConfirmModal: () => void;
+}>({
+  openConfirmModal: async () => undefined,
+  closeConfirmModal: () => undefined,
+});
+
+export const SignProvider = ({ children }: { children: ReactNode }) => {
+  const { t } = useTranslation();
+
+  const account = usePopupSelector(state => state.selectedAccount);
+
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const modalPromiseRef = useRef<{
+    resolve: null | ((value: unknown) => void);
+  }>({ resolve: null });
```

Comment:
Seems like making the whole object nullable makes more sense. Or may be not use an object, just keep resolve fn itself and name it `modalResolveRef`